src/kudu/consensus/consensus_queue.h

```diff
@@ -368,7 +368,7 @@ class PeerMessageQueue {
 
   ~PeerMessageQueue();
 
-  // Begin or end the watch for an eligible successor. If 'successor_uuid' is
+  // Begin or end the watch for an eligible successor. If 'successor_uuid' is not
```

Comment:
this comment was correct previously. Its a slightly roundabout way they have implemented how they chose a new leader or use the one that has been suggested.

Reply:
what it is saying is that if successor_uuid is null, then we let the code fall
through and chose an eligible successor

Reply:
this is like the case of a transfer leadership with no "fixed" destination. Any caught up follower is fine.